### PR TITLE
GODRIVER-1858 Use mock deployment for slow BulkWrite test

### PR DESCRIPTION
**Pull request by divjotarora**
Based on local logging, it seems like most of the time was spent deleting documents server-side. Switching to the mock deployment speeds up the test from ~57s to 0.25s. 

In addition to using the mock deployment for a specific test, this PR also modifies the mock deployment code to export a server description. This allows us to programatically determine the required batch size for a test without having to assume it's 100K. I also chose to export the `SupportedWireVersions` variable in the `topology` package so the mock deployment could access the max supported wire version rather than hardcoding it, which avoids the risk of it getting outdated.

### CODE REVIEW

File: x/mongo/driver/topology/fsm.go [hunk 1]
```diff
@@ -16,8 +16,12 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/description"
 )
 
-var supportedWireVersions = description.NewVersionRange(2, 9)
-var minSupportedMongoDBVersion = "2.6"
+var (
```
Comment (iwysiu) [inline x/mongo/driver/topology/fsm.go:19]: Should these be const?

Comment (divjotarora) [reply x/mongo/driver/topology/fsm.go:19]: `SupportedWireVersions` cannot be because it's a struct but I changed `MinSupportedMongoDBVersion`.

File: mongo/integration/change_stream_test.go [hunk 2]
```diff
@@ -236,6 +236,7 @@ func TestChangeStream_ReplicaSet(t *testing.T) {
 			Code:    errorHostUnreachable,
 			Name:    "foo",
 			Message: "bar",
+			Labels:  []string{"ResumableChangeStreamError"},
```
Comment (benjirewis) [inline mongo/integration/change_stream_test.go:239]: How is this related to other changes?

**Review by benjirewis — APPROVED** [review]
LGTM! 1m to .25s is awesome 😮

**Review by iwysiu — APPROVED** [review]

File: mongo/integration/change_stream_test.go (hunk 2 above)
Comment (divjotarora) [reply mongo/integration/change_stream_test.go:239]: I switched the mock deployment to use the latest wire version we support (9, or server version 4.4), which was an upgrade from the existing reported version of 8 (or server version 4.2). For server versions >= 4.4, change streams consider a server error to be resumable if it has the `ResumableChangeStreamError` label, so this test needed to be modified to include this label.

File: mongo/integration/collection_test.go [hunk 3]
```diff
@@ -1661,9 +1661,10 @@ func TestCollection(t *testing.T) {
 				mt.Fatalf("expected BulkWrite error %v, got %v", mongo.ErrUnacknowledgedWrite, err)
 			}
 		})
-		mt.Run("insert and delete with batches", func(mt *mtest.T) {
+		mt.RunOpts("insert and delete with batches", mtest.NewOptions().ClientType(mtest.Mock), func(mt *mtest.T) {
 			// grouped together because delete requires the documents to be inserted
-			numDocs := 100050
+			maxBatchCount := int(mtest.MockDescription.MaxBatchCount)
```
Comment (kevinAlbs) [inline mongo/integration/collection_test.go:1666]: No need to change, but IIUC, `MaxBatchCount` could be set to another value at this point. Monitoring and connection handshake is bypassed for the mock deployment, so this won't be checked until the operation runs.

**Review by kevinAlbs — APPROVED** [review]
LGTM, the mock deployment is pretty neat.

File: mongo/integration/collection_test.go (hunk 3 above)
Comment (divjotarora) [reply mongo/integration/collection_test.go:1666]: Yeah, it can be. Reducing the batch size would've been my next step if the test was still taking too long, but we got significant improvement already and I didn't feel the need to optimize further.
